bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java

```diff
@@ -295,7 +307,9 @@ Iterable<Table> listTables(DatasetId datasetId, TableDefinition.Type... types) {
   TableId createDestinationTable(
       Optional<String> referenceProject, Optional<String> referenceDataset) {
     String project = materializationProject.orElse(referenceProject.orElse(null));
-    String dataset = materializationDataset.orElse(referenceDataset.orElse(null));
+    String dataset =
```

Comment:
Open question here. If customers do not pass in a dataset while pushing down a query, do we create one for them?

Here, I have added code that creates a dataset if none is given

Reply:
Synced with David. We do not want to create the dataset ourselves for now